process.go

```diff
@@ -12,40 +12,43 @@ import (
 
 // LaunchProcess runs a subprocess and returns when the subprocess exits,
 // either when it dies, or *after* a successful upgrade.
-func LaunchProcess(cfg *Config, args []string, stdout, stderr io.Writer) error {
+func LaunchProcess(cfg *Config, args []string, stdout, stderr io.Writer) (bool, error) {
+	var doUpgrade bool
 	bin := cfg.CurrentBin()
+
 	err := EnsureBinary(bin)
 	if err != nil {
-		return errors.Wrap(err, "current binary invalid")
+		return doUpgrade, errors.Wrap(err, "current binary invalid")
```

Comment:
I find this confuing, as you must scan all the code above this point to understand if it is true or false.
Why not just use a constant everywhere - eg, `return false, errors.Wrap(err, "current binary invalid")`